hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/CallFeature.java

```diff
@@ -291,12 +357,325 @@ public void ethCallReentrancyCallFunction() {
+        // totalSupplyBefore - amount = totaSupplyAfter
+        assertThat(results.get(2) - 1L).isEqualTo(results.get(3));
+    }
+
+    // ETHCALL-074
+    @Then("I wipe NFT token and return the total supply and balance of treasury")
+    public void ethCallWipeNftTokenGetTotalSupplyAndBalanceOfTreasury() {
+        var data = encodeData(
+                PRECOMPILE,
+                WIPE_TOKEN_GET_TOTAL_SUPPLY_AND_BALANCE_OF_TREASURY_SELECTOR,
+                asAddress(nonFungibleTokenId.toSolidityAddress()),
+                0L,
+                asLongArray(List.of(1L)),
+                asAddress(receiverAccountAlias));
+
+        var response = callContract(data, precompileContractAddress);
+        balanceOfToken = getBalanceOfToken(nonFungibleTokenId, receiverAccountId.getAccountId());
+        totalSupplyOfToken = getTotalSupplyOfToken(nonFungibleTokenId);
+        var results = response.getResultAsListDecimal();
+
+        // BalanceBefore - amount = balanceAfter
+        assertThat(results.get(0) - 1).isEqualTo(results.get(1));
+        // totalSupplyBefore - amount = totalSupplyAfter
+        assertThat(results.get(2) - 1).isEqualTo(results.get(3));
+    }
+
+    // ETHCALL-075, ETHCALL-076
+    @Then("I call function that pauses {string} token gets status unpauses and returns the status of the token")
+    public void ethCallPauseTokenGetStatusUnpauseGetStatus(String tokenName) {
+        var tokenId = tokenClient
+                .getToken(TokenClient.TokenNameEnum.valueOf(tokenName))
+                .tokenId();
+        var data = encodeData(PRECOMPILE, PAUSE_GET_STATUS_UNPAUSE_GET_STATUS, asAddress(tokenId));
+        var response = callContract(data, precompileContractAddress);
+        var statusAfterPause = response.getResult().substring(2, 66);
+        var statusAfterUnpause = response.getResult().substring(66);
+
+        // pause status is true
+        assertThat(Integer.valueOf(statusAfterPause, 16)).isEqualTo(1);
+        // unpause status is false
+        assertThat(Integer.valueOf(statusAfterUnpause, 16)).isEqualTo(0);
+    }
+
+    // ETHCALL-077, ETHCALL-078
+    @Then("I call function that freezes {string} token gets freeze status unfreezes and gets freeze status")
+    public void ethCallFreezeFungibleGetFreezeStatusUnfreezeGetFreezeStatus(String tokenName) {
+        var tokenId = tokenClient
+                .getToken(TokenClient.TokenNameEnum.valueOf(tokenName))
+                .tokenId();
+        var data = encodeData(
+                PRECOMPILE, FREEZE_TOKEN_GET_STATUS_UNFREEZE_GET_STATUS, asAddress(tokenId), asAddress(admin));
+        var response = callContract(data, precompileContractAddress);
+        var statusAfterFreeze = response.getResult().substring(2, 66);
+        var statusAfterUnfreeze = response.getResult().substring(66);
+
+        // status after freeze is true
+        assertThat(Integer.valueOf(statusAfterFreeze, 16)).isEqualTo(1);
```

Comment:
Use the SDK proto enum then it is self documenting. Same comment elsewhere.

```suggestion
        assertThat(Integer.valueOf(statusAfterFreeze, 16)).isEqualTo(TokenPauseStatus.Paused.getNumber());
```

Reply:
Hey Steven, for most of these enums we have 3 enum values - for the case of pause, those are `PAUSED`, `UNPAUSED`, `NOT_APPLICABLE` with 3 different enum indexes.

The precompile however works with boolean values and we expect one of two values - `true` for `PAUSED` and `false` for `UNPAUSED` and `NOT_APPLICABLE`.

I believe it's better to check the returned values following users perspective and expect only `true` or `false` values.

Reply:
Ok, bad assumption on my part.